torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1298,3 +1352,25 @@ def _alloc_storage(data: torch.Tensor, size: torch.Size) -> None:
         data.storage().size() == 0
     ), "Then tensor storage should have been resized to be 0."
     data.storage().resize_(size.numel())  # type: ignore[attr-defined]
+
+
+def _calc_grad_norm(parameters: List[torch.nn.Parameter], p: float) -> torch.Tensor:
+    r"""Calculate gradient norm of an iterable of parameters.
+    Returns:
+        Total norm of the parameters (viewed as a single vector).
+    """
+    if isinstance(parameters, torch.Tensor):
+        parameters = [parameters]
```

Comment:
`parameters` is typed as a List[Parameter], are there cases we can just get a single tensor?

Reply:
That's a good point, this is also copied from FairScale. Let me remove this check.